GetandTakeAPI/Controllers/CategoriesController.cs

```diff
@@ -1,30 +1,243 @@
     public CategoriesController(ICategoryService categoryService)
     {
         _categoryService = categoryService;
     }
 
-    [HttpGet]
-    public async Task<ActionResult<List<Category>>> GetAllCategoriesAsync()
+    /// <summary>
+    /// Gets All Categories
+    /// </summary>
+    /// <returns>
+    /// A <see cref="Task"/> representing the asynchronous operation with List of <see cref="CategoryResponse"/>
+    /// </returns>
+    /// <response code="200"> Category List has been found.</response>
+    [HttpGet(Name = nameof(GetAllCategoriesAsync))]
+    [ProducesResponseType(typeof(CategoryResponse), StatusCodes.Status200OK)]
+    public async Task<ActionResult<List<CategoryResponse>>> GetAllCategoriesAsync()
     {
         var categories = await _categoryService.GetAllAsync();
-        
+
         if (categories == null)
         {
             return BadRequest();
         }
 
         return categories;
     }
+
+    /// <summary>
+    /// Returns information about Category by id.
+    /// </summary>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST / Categories/id/{id}
+    ///     {
+    ///       "id":0
+    ///     }
+    ///
+    /// </remarks>
+    /// <param name="id">to take selected category by Category identifier</param>
+    /// <returns>
+    /// A <see cref="Task"/> representing the asynchronous operation with <see cref="CategoryResponse"/>
+    /// </returns>
+    /// <response code="200"> Category has been found.</response>
+    /// <response code="404"> Unable to find Category.</response>
+    [HttpGet("id/{id:int}")]
+    [ProducesResponseType(typeof(CategoryResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<CategoryResponse>> GetCategoryByIdAsync(int id)
+    {
+        var category = await _categoryService.GetByIdAsync(id);
+
+        if (category == null)
+        {
+            return BadRequest();
+        }
+
+        return category;
+    }
+
+    /// <summary>
+    /// Creates a Category
+    /// </summary>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST / Categories
+    ///     {
+    ///         "categoryName": "string",
+    ///         "description": "string",
+    ///         "imagePath": "string"
+    ///     }
+    ///
+    /// </remarks>
+    /// <param name="category"><see cref="CategoryDetail"/></param>
+    /// <returns>    
+    /// A <see cref="Task"/> representing the asynchronous operation with <see cref="CategoryDetail"/>
+    /// </returns>
+    /// <response code="201">Returns the newly created CategoryDetail</response>
+    /// <response code="500">Unable to create Category due to internal issues.</response>
+    /// <response code="400">If the item is null</response>
+    [HttpPost]
+    [ProducesResponseType(typeof(CategoryDetail), StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
```

Comment:
Let's use `typeof(void)` to make sure that Swagger does not suggest some wrong models in response